STN-336 :  QA Feedback Sprint 15

heatherdesigns opened the pull request:
#  🛑 [STN-336](https://sparkbox.atlassian.net/browse/STN-336) DO NOT MERGE - SEE new PR #592

## Summary
Implement fixes for items discovered during during the Sprint 15 QA Feedback session.

The work in this PR has been browser tested. ✅ 

## Need Review By (Date)
5/6/2020

## Urgency
low

## Steps to Test
1. In the CLI, run `npm run test` and confirm all our Sass tests pass.
2. Confirm each of the following items have been resolved.


- [x] External Link icon is inconsistently applying on WYSIWYG
    * This was resolved in QA Sprint 14 Feedback PR. Confirm by visiting the QA Real Econ Content page which was built using the WYSIWYG. http://economics.suhumsci.loc/qa/qa-real-econ-content
<br>

- [x] Prevent external links (and external link styling) from appearing in the Drupal Administration Toolbar
    * I was unable to duplicate this issue locally, but you can see it on the dev environment for archaeology.
    * In the CLI, run `lando drush @archaeology.dev uli` to create a login URL for the archaeology site. Copy and paste the URL provided into your web browser. Eg. `http://archaeology-dev.stanford.edu/user/reset/1/1588259656/aG-T70_eSMOy5wezcVEagQHUBraFVzBZiOXu_5Vwdrk/login` 
    * Confirm the external link icon is gone (as a result of the work in QA Feedback Sprint 14)
  * _hmm I still see something (an arrow) on interior pages_
    * Open the inspector tools. Find the anchor link for the `Subscription active` tab. Confirm that setting the `background-image` to `none` will resolve the external link styling issue. ✅ 
    * The database settings which exclude external links from displaying in particular locations have also been updated to include the toolbar administration.
<br>

- [x] External Link icon applying to postcards and breaking arrow link styling
    * NOTE: The styles causing this bug were removed in the Sprint 14 QA Feedback PR
    * To confirm, go to http://economics.suhumsci.loc/qa/qa-cards
    * Click on a contextual link to edit the page
    * Edit a postcard
    * Update the READ MORE LINK to an external link such as `http://google.com` and Save
<br>

- [x] Email links apply envelope on postcards
    * Go to http://economics.suhumsci.loc/qa/qa-cards
    * Click on a contextual link to edit the page
    * Edit a postcard
    * Update the READ MORE LINK to a mail link such as `mailto: abc@xyz.com` and Save
    * Confirm the link on the postcard does not have an email envelope icon to the right of the arrow icon
<br>

- [x] Email links in views aren't styled appropriately
    * Go to http://economics.suhumsci.loc/people/faculty
    * Confirm email icon is present, the text and icon are the right color, and the icon changes color when hovering over the link text and the icon _hmm I dont see the icon color changing_. 
    * You can also see what a link with an email icon looks like within a sentence on this page: http://economics.suhumsci.loc/seminars-workshops/list-seminars. Look at the fourth item of `Development` in the list.
<br>

- [x] Exposed Filter in Sidebar
    * Go to http://economics.suhumsci.loc/courses
    * Click on `Layout` in the Drupal Admin bar below the site navigation
    * In the left hand sidebar of `Section 2`, click `+ Add block`
    * Under `Courses (Views)` select `Custom Econ Courses: All List Exposed Form` and `Add block`
    * Save the layout and go back to the courses page
<br>

- [x] Full pager Styling
    * Go to http://economics.suhumsci.loc/about/news
    * Scroll to the bottom of the page. The full pager style should already be implemented.
    * Look at the `active` and `hover` state background circle. It should look like a circle when there is only 1digit and like a button when there are 2 or more digits.
<br>

- [x] Mini pager Styling
    * Go to http://economics.suhumsci.loc/about/news
    * Click on the contextual link to edit the `View`
    * Under `Block Setting` find `Pager`, under `Use pager`, click on `Full`
    * Update to `Paged output, mini pager` then Apply and Save
<br>

- [x] Image floating/wrapping in text areas
    * Create a basic page and add text areas that include headings, images floated to the right and left, ordered and unordered lists, and body copy
    * Verify the following issues have been resolved:
        * When an image is floated to the right alongside a list, the bullets align beside the list as expected.
        * When an image floats left or right in a text area component, the content in a component placed below does not wrap around the floated image. (Ensure floats are cleared after each row.)
        * In the text area editor, make sure you are always able to `edit` an image. There was an issue with not being able to click on the `edit` button for an image when it was floated alongside lists.
<br>

- [x] Event Series Menu Block
    * Go to http://economics.suhumsci.loc/seminars-workshops/list-seminars and to http://economics.suhumsci.loc/events/public-economics-and-environmental-economics
    * Verify the sidebar on the Public Economics and Environmental Economics page is the same width as the sidebar on the List of Seminars page
<br>

- [x] View Titles get an odd small styling
    * Go to http://economics.suhumsci.loc/seminars-workshops/upcoming-seminars-and-events
    * I _think_ this may have been resolved in the work on headings in the QA Feedback for Sprint 14. Take a look and see if you agree. 👍 
<br>


## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

Commit:
fix(STN-315): word break issue (#567)

- Revert to default Decanter behavior as I think that's for the best

Commit:
feat: add Hi-Res support to WYSIWYG images styles (#568)

JIRA STN-259
https://sparkbox.atlassian.net/browse/STN-259

kaseybon committed:
feat: sprint 14 qa feedback (#571)

- Created hb-title utility
- Fixed button classes when used on a wrapper field
- Removed default external link icon, it can now be added with a class
- Fixed heading styles when alignment classes are added
- Updated hs-font-lead styles to be more distinctive

STN-319

bryanbraun committed:
feat(STN-289): prevent menu flash during page-load (#574)

Our menus are expanded by default so users with JavaScript disabled can
navigate the site. This was contributing to a distracting flash of
expanded menus during the page load.

The solution was two-fold:

1. Preventing the flash of "no-js" styles, by removing the no-js class
   earlier (in the head of the document, instead of at the bottom).

2. Preventing the flash of expanded menus, by executing the "menu
   collapse JavaScript" immediately after the DOM nodes are added to the
   document.

Commit:
feat(STN-322) : More Link work (#578)

* feat: style the more link

* fix: use hs-button for infinate scroll button

* refactor: reset button.hs-more-link default styles, use animation vars in all transitions

* feat: add compiled CSS

Co-authored-by: Kasey Bonifacio <kasey@heysparkbox.com>

Commit:
feat(STN-306): background color for Row Paragraph type (#577)

* feat(STN-306): adds new field for row paragraph type styles

- adds new config files for DB changes
- adds hook logic to add classes to DOM
- adds a couple preliminary styles for new row styles

* feat(STN-306): adds more well paragraph styles

* fix(STN-306): removes unecessary package entity_class_formatter

- I thought this could be a potential soln, but we ended up not needing
  it

* feat(STN-306): PR feedback:

- extends hb-well
- adds period to php comment
- adds compiled css

Commit:
feat(STN-324) : Detail Page Image Styles (#576)

* feat: create image styles utility class

* feat: implement design feedback

- adjust width property
- add margin for larger screen sizes

* feat: experiment with removing widths on the title and image for detail node pages

- remove utility classes that we have decided not to use
- clean up / remove comments
- refine styles for node detail pages

* feat: add comment about how image styles are handled

* feat: add complied CSS

kaseybon committed:
fix(STN-321): resolve IE JS errors, refactor secondary nav to use custom JS (#580)

Commit:
STN-157 : Date Stacked Horizontal UI Pattern (#570)

* feat: create date stacked horizontal card UI pattern

JIRA STN-157
https://sparkbox.atlassian.net/browse/STN-157

* feat: apply date stacked horizontal card styles

* feat: adjust pattern to not render drupal output when field data is empty

* feat: implement feedback from design pairing

* fix: IE alignment bug

- replace the use of postion: initial with position: static
- IE doesn't support position: initial and was causing the content to stack vertically rather than display horizontally

* fix: linting error for empty line rule

* feat: add calendar icon and styles for the date and time label and field

* feat: add location and speaker icons and their styles

* feat: update class name to be more reflective of use case

* feat: scope icons to an hb-card__icon class

- this creates flexibility should we decide to implement icons in other templates that use the hb-card classes

* feat: add compiled CSS after feedback changes

Commit:
feat: implement Sprint 15 QA feedback

- hide email link icons on postcards

Commit:
feat: prevent external links (and external link styling) from appearing in the Drupal Administration toolbar

Commit:
feat(STN-337):  Humsci Traditional Theme work (#579)

* feat(STN-337): adds cardinal theme with copy of Sand colorful pairing

* feat(STN-337): adds cardinal theme setting and config

* feat(STN-337): adds airy to colorful styles where traditional differ

* feat(STN-337): adds link styles, and deletes colorful specific colors to allow for global colors on all themes

* feat(STN-337): adds tradl button styles

* refactor(STN-337): removes humsci colorful specific colors where possible so they apply to themes more globally. 
- I didnt address heros though since those feel bigger and Im not sure what the default should be since they seem different between themes

* feat(STN-337): adds more proper link styles

* feat(STN-337): adds new global grays for traditional and linting fix

* refactor(STN-337): refactor background image stuff in link text mixins

* fix(STN-337): PR feedback, some refactor, and notes for future refactor

* feat(STN-337): adds compiled CSS

Commit:
feat: reset background-image value for external links in the admin toolbar

Commit:
feat: fix email link styling

Commit:
feat: update hb-link-background-image to use the new icon-width variable to set the amount of space which the icon needs to occupy

Commit:
feat: fix spacing for exposed filter when placed in the sidebar

Commit:
feat: adjust hover state of email links in layout builder where email icons are not being used

Commit:
feat: allow exposed form reset button hover state to recieve the styles as written by adding !important

Commit:
refactor: views exposed form after discovering that the styles broke the search form in the header

Commit:
feat: refine pager styles active and hover states and add styles for the mini pager

Commit:
feat: add clearfix and flexbox styles to fix image floating / wrapping issues

Commit:
feat: update three column with image layout sidebar width to match the same settings as the three column layout

Commit:
feat: browser testing fixes

- add property to exposed form styles to allow inputs to display at correct width

silverli reviewed (changes requested):
On initial review, there are a couple things I wasn't able to verify:

- archeology subscription page. looks like there is still an arrow if you go to an internal admin page, like create a page
- is this what the mini pager should look like? Maybe I need to adjust another setting:
<img width="404" alt="Screen Shot 2020-05-04 at 2 44 14 PM" src="https://user-images.githubusercontent.com/12848168/81003421-ddd05a00-8e18-11ea-851f-ec3c28d42c0a.png">

- on http://economics.suhumsci.loc/people/faculty I don't see that the email icon changes color on hover
- view titles are enormous on mobile. but I know merani wants to work on these as part of polish work anyways so maybe it's part of that. look at it and let me know what you think!
<img width="530" alt="Screen Shot 2020-05-04 at 3 03 36 PM" src="https://user-images.githubusercontent.com/12848168/81003434-e2950e00-8e18-11ea-8ad9-4eca6d77e462.png">

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/utilities/_toolbar.scss:
```diff
@@ -3,6 +3,10 @@
       background-image: none;
+
+      &:hover {
+        background-image: linear-gradient(rgba(255, 255, 255, 0.125) 20%, transparent 200%); // reset the default background-image styles for the toolbar
```
where does this style come from?

heatherdesigns replied:
This is the hover state of items in the toolbar. By applying `background-image: none` to the anchor link to override our `background-image` link styles, I overwrote Drupal's hover state styles that also use `background-image`. So I added them back.

silverli replied:
hmm can you write a comment about that and how colorful theme is what is causing the issue?

heatherdesigns replied:
Yep! 😄

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_views-exposed-form.scss:
```diff
@@ -1,4 +1,4 @@
-.hb-layout-builder-main-content .views-exposed-form {
+.views-exposed-form { // TODO: ask Merani about Reset button hover state
```
@heatherdesigns did you get a chance to ask Merani?

heatherdesigns replied:
I didn't ask because I saw where the styles were there but not displaying until I added `!important`. 🤷

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_views-exposed-form.scss:
```diff
@@ -7,25 +7,40 @@
-    @include grid-media-min('lg') {
-      width: calc(33% - #{hb-calculate-rems(18px)});
+
+    .hb-layout-builder-main-content & {
```
hmm is this the right class to target? maybe this is better? `hb-three-column__main`

heatherdesigns replied:
That makes sense to me! The `hb-layout-builder-main-content` class was used before so that's why I re-used it. But this makes more sense as `hb-three-column__main` and `hb-three-column__sidebar-1` sit beside each other in the DOM.

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_views-exposed-form.scss:
```diff
@@ -12,35 +13,36 @@
 
-    &:not(:last-child) {
-      margin-right: hb-calculate-rems(16px);
+.hb-layout-builder-main-content .views-exposed-form {
```
again, not sure this is the right class to use.. I think `hb-three-column__main` is better in all instances unless there's another reason

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pager.scss:
```diff
@@ -1,9 +1,32 @@
+// Base pager styles
```
are these actually for mini pager? not the base styles?

silverli replied:
is there a way to write these styles and share styles across the mini pager and full one more?

silverli replied:
(basically DRY-ing up logic that is used here twice in mini pager and full pager)

heatherdesigns replied:
I removed my comment of `// pager__item styles are written below` as I think it was probably misleading. 

This one was a challenge due to the full pager having classes I could style, but the mini pager doesn't. I was happy to originally get to the point of not using any `!important`. 😆It was pretty gnarly at one point. But I'm glad you called this out as I was able to take a fresh look at it and hopefully make it even better. :tada:

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.general.scss:
```diff
@@ -52,6 +52,8 @@
 }
 
 @mixin hb-unordered-list {
+  display: flex;
```
what does this do?

heatherdesigns replied:
The combination of `display: flex` used with `flex-direction: column` allow the bullets and numbers on unordered and ordered lists to align correctly when lists are used alongside floated images. Honestly, I was struggling to figure this one out and then thought, hmmm..... I wonder what flexbox would do and :boom: it worked! But I'm open to other ideas. 💡 

Here's a screenshot of before:

![before](https://user-images.githubusercontent.com/12678977/81016962-6eb23000-8e2f-11ea-9d43-f314ea956858.png)

And a screenshot of after:

![after](https://user-images.githubusercontent.com/12678977/81016971-72de4d80-8e2f-11ea-82a2-3ff2e5364c68.png)

silverli replied:
ok! that makes sense, I guess I am now  just wondering if we should only applyy display flex to lists when they are in a wysiwyg textarea, and not all the time. Maybe issues could arise if that's always applied.

heatherdesigns replied:
Yes!!! I removed the the flexbox styles from the mixins and added them to the WYSIWYG to narrow the scope. Good call!

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/utilities/_wysiwyg-text-area.scss:
```diff
@@ -2,6 +2,7 @@
 
 .field-hs-text-area {
   margin-bottom: hb-calculate-rems(48px);
+  @include clearfix;
```
👍

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss:
```diff
@@ -38,7 +38,7 @@
       .hb-three-column-w-image--one-sidebar &,
       .hb-three-column-w-image--two-sidebar & {
-        width: 100%;
+        width: calc(100% - #{$hb-sidebar-width} - #{hb-calculate-rems($hb-gutter-width)});
```
lol good catch

silverli replied:
hmm but if there's two sidebars, I wonder if the logic is still messed up

silverli replied:
I think I checked and it looks ok on different breakpoints, but it'd be great if you could check too!

heatherdesigns replied:
I double checked. Looks good to me! :eyes:

heatherdesigns commented:
## Review Feedback
* Good catch on the arrow icon on the archeology page! I added a `display:none` to hide it.
* Yes! That is the mini pager styling. :smile:
* I had the email icons wrapped in the `hb-colorful` mixin so you couldn't see them if you were in traditional. I realized that wasn't necessary and removed the `hb-colorful` mixin so you can see the correct icon colors across all themes.
* Regarding the view titles, I'll schedule time with Merani to review.

Commit:
feat: implement review feedback

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pager.scss:
```diff
@@ -1,13 +1,19 @@
     margin: hb-calculate-rems(10px);
+  }
 
+  li:not(.pager__item),
```
so much better! thanks for consolidating!

silverli reviewed (changes requested):
this is really close! there are a couple more comments I left on things from yesterday where I'd like some extra documentation added, etc.

Also, I know you still are talking to merani about the headings as titles on pages - let me know where that ends up.

Lastly, I did notice that the email icon shows up in the vertical linked card:

<img width="327" alt="Screen Shot 2020-05-05 at 12 28 08 PM" src="https://user-images.githubusercontent.com/12848168/81091843-c606dd80-8ecd-11ea-83c9-10805e5fb2a7.png">

Commit:
feat: implement round 2 feedback

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.general.scss:
```diff
@@ -52,8 +52,6 @@
 }
 
 @mixin hb-unordered-list {
-  display: flex;
```
thanks for making this change!

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/utilities/_toolbar.scss:
```diff
@@ -1,4 +1,5 @@
-// Drupal Administration Toolbar overrides
+// Drupal Administration Toolbar
+// Override anchor link styles written for the Humsci Colorful theme.
```
Lovely!

silverli reviewed (approved):
Great work!! let's squash everything and make a new PR rebased on the sprint branch. 🎉

silverli commented:
Closing in favor of newly rebased branch here: https://github.com/SU-HSDO/suhumsci/pull/592
